src/phylum/ci/ci_base.py

```diff
@@ -106,19 +78,74 @@ def __init__(self, args: Namespace) -> None:
+        return self._args
 
-    @property
-    def is_lockfile_changed(self) -> bool:
-        """Get the lockfile's modification status."""
-        return self._lockfile_changed
+    @cached_property
+    def lockfiles(self) -> Lockfiles:
+        """Get the package lockfile(s) in lexicographic order.
+
+        The package lockfile(s) can be specified as an option or contained in the `.phylum_project` file.
+        Lockfiles provided as an input option will be preferred over any entries in the `.phylum_project` file.
+
+        When no valid lockfiles are provided otherwise, an attempt will be made to automatically detect them.
+        """
+        arg_lockfiles: Optional[List[List[Path]]] = self.args.lockfile
+        if arg_lockfiles:
+            # flatten the list of lists
+            provided_arg_lockfiles = [path for sub_list in arg_lockfiles for path in sub_list]
+            print(f" [+] Lockfile(s) provided as arguments: {provided_arg_lockfiles}")
+            valid_lockfiles = self.filter_lockfiles(provided_arg_lockfiles)
+            if valid_lockfiles:
+                print(f" [-] Valid provided lockfiles: {valid_lockfiles}")
+                return valid_lockfiles
+
+        print(" [+] No valid lockfiles were provided as arguments. Checking the `.phylum_project` file ...")
+        lockfile_entries: List[OrderedDict] = self._project_settings.get("lockfiles", [])
+        lockfile_paths = [lockfile_entry.get("path") for lockfile_entry in lockfile_entries]
+        provided_project_lockfiles = [Path(lockfile) for lockfile in lockfile_paths if lockfile]
+        if provided_project_lockfiles:
+            print(f" [+] Lockfile(s) provided in `.phylum_project` file: {provided_project_lockfiles}")
+            valid_lockfiles = self.filter_lockfiles(provided_project_lockfiles)
+            if valid_lockfiles:
+                print(f" [-] Valid provided lockfiles: {valid_lockfiles}")
+                return valid_lockfiles
+
+        print(" [+] No valid lockfiles in the `.phylum_project` file. An attempt will be made to detect lockfiles.")
+        detected_lockfiles = detect_lockfiles()
+        if detected_lockfiles:
+            print(f" [+] Detected lockfile(s): {detected_lockfiles}")
+            valid_lockfiles = self.filter_lockfiles(detected_lockfiles)
+            if valid_lockfiles:
+                print(f" [-] Valid detected lockfiles: {valid_lockfiles}")
+                return valid_lockfiles
+
+        raise SystemExit(" [!] No valid lockfiles were detected. Consider specifying at least one with `--lockfile`.")
+
+    def filter_lockfiles(self, provided_lockfiles: List[Path]) -> Lockfiles:
+        """Filter potential lockfiles and return the valid ones in sorted order."""
+        lockfiles = []
+        for provided_lockfile in provided_lockfiles:
+            if not provided_lockfile.exists():
+                print(f" [!] Provided lockfile does not exist: {provided_lockfile}")
+                continue
```

Comment:
It seems odd not to raise an error here. If a user configures `phylum-ci` to look at `pakcage-lock.json` (sic) and no file exists by that name, a hard error is a good way to alert the user to their mistake.

Reply:
The change has been made and tested:

<img width="1449" alt="image" src="https://user-images.githubusercontent.com/18729796/228329976-aa170323-d0c6-4c5d-8e5a-717d60fbbc47.png">